crates/common/mqtt_channel/src/connection.rs

```diff
@@ -166,6 +170,27 @@ impl Connection {
                     let _ = message_sender.send(msg.into()).await;
                 }
 
+                Ok(Event::Incoming(Packet::ConnAck(ack))) => {
+                    // If session_name is not provided, then re-subscribe
+                    if config.session_name.is_none() {
+                        eprintln!("INFO: Connected");
+                        if let Some(err) = MqttError::maybe_connection_error(&ack) {
+                            eprintln!("ERROR: Connection Error {}", err);
+                        };
+                        let subscriptions = config.subscriptions.filters();
+                        if subscriptions.is_empty() {
+                            break;
+                        }
+                        match mqtt_client.subscribe_many(subscriptions).await {
+                            Ok(()) => {}
+                            Err(e) => {
+                                eprintln!("ERROR: Failed to subscribe due to {}", e);
+                                break;
+                            }
+                        }
```

Comment:
This piece of code have been copy/pasted from the `open` function. Please, introduce a `subscribe` function that takes as argument the `mqtt_clients` and the `subscriptions` vector.

Reply:
Introduced a `subscribe` function to get rid of redundant code.